ocaml/testsuite/tests/typing-layouts-or-null/reexport.ml

```diff
@@ -0,0 +1,234 @@
+(* TEST
+ flags = "-extension layouts_alpha";
+ expect;
+*)
+
+(* CR layouts v3: ['a or_null] can't be re-exported normally,
+   because users can't define their own [Null]-like constructors. *)
+module Or_null = struct
+  type ('a : value) t : value_or_null = 'a or_null =
+    | Null
+    | This of 'a
+end
+
+[%%expect{|
+Lines 2-4, characters 2-16:
+2 | ..type ('a : value) t : value_or_null = 'a or_null =
+3 |     | Null
+4 |     | This of 'a
+Error: The kind of type 'a or_null is value_or_null
+         because it is the primitive value_or_null type or_null.
+       But the kind of type 'a or_null must be a subkind of value
+         because of the definition of t at lines 2-4, characters 2-16.
```

Comment:
This is a poor error message. It should say something about the fact that variants are values. Doesn't necessarily have to be fixed in this patch, but if not, leave a CR to return to this.